### PR TITLE
INFRA-701 Kill all the orphans

### PR DESCRIPTION
**Pull request by dfuentes**
https://clever.atlassian.net/browse/INFRA-701

Some aviator services were being run indirectly by "go run" and killing the go run process was not killing its child.

This change uses setsid to ensure all child processes are started in their own process group. We then call send a kill signal to -pid, which is the process group id for a the process with id pid.

Since we are using syscall package, this ties us to os's that support these features.  I have tested on linux and os x.

### CODE REVIEW

File: aviator/servicelist.go [hunk 1]
```diff
@@ -135,16 +140,10 @@ func (s *ServiceList) shutdown() error {
...
-			}
-			// Wait for it to actually die
-			_, err = service.cmd.Process.Wait()
-			if err != nil {
-				return err
-			}
+			// Sending a kill signal to -pid kills the process group
+			// for the process pid
+			syscall.Kill(-pid, syscall.SIGKILL)
+			service.cmd.Process.Wait()
```
Comment (nathanleiby) [inline aviator/servicelist.go]: Is it safe to ignore the errors now?

Comment (dfuentes) [inline aviator/servicelist.go]: It was safe before as well.  This only throws an error for a race condition (the process existed when we did isProcessRunning, but died before we sent the kill signal), in which case we can ignore the error.

**Comment by nathanleiby** [comment]
lgtm

**Comment by nathanleiby** [comment]
Would be great to have a unit test, but agreed the setup is a bit complex.

**Pull request merged**